### PR TITLE
Task/39 build test scripts csclsroz 14

### PR DESCRIPTION
**Pull request by DougMidgley**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

_Please delete options that are not relevant._

- [ ] Documentation update
- [ ] Bug fix
- [ ] New metadata support
- [ ] Enhanced metadata
- [ ] Add a CLI option
- [X] Add something to the core
- [ ] Other, please explain:

## What changes did you make? (Give an overview)

Test cases for Query & Data Extension to help improve early detection of issues

## Is there anything you'd like reviewers to focus on?

...

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] ESLint & Prettier are not outputting errors or warnings
- [X] README.md updated (if applicable)
- [X] Release Notes Updates

### CODE REVIEW

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]
dude - tons of work in this one. my respect! 
nevertheless, got a few things that need clearing up and then a demo (or quick pointer on how to run it) would be in order

File: test/mockRoot/.mcdev-auth.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "credentials": {
+        "testInstance": {
+            "clientId": "71fzp43cyb1aksgflc159xxx",
+            "clientSecret": "oDZE354QsMXzcFqKQlGgDxxx",
+            "tenant": "mct0l7nxfq2r988t1kxfy8sc4xxx",
+            "eid": 1111111
+        }
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply test/mockRoot/.mcdev-auth.json:8]: obfuscated yes, but can put a full dummy value if you prefer :)

File: test/dataExtension.test.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const assert = require('chai').assert;
+const cache = require('../lib/util/cache');
+const testUtils = require('./utils');
+const handler = require('../lib/index');
+
+describe('retrieve', () => {
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply test/dataExtension.test.js:6]: npm run test, as per package.json :)

File: test/mockRoot/deploy/testInstance/testBU/dataExtension/childBU_dataextension_test.dataExtension-meta.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "Fields": [
+        {
+            "PartnerKey": "",
+            "ObjectID": "",
+            "Name": "testField",
+            "DefaultValue": "",
+            "MaxLength": 254,
+            "IsRequired": true,
+            "IsPrimaryKey": false,
+            "FieldType": "Text"
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply test/mockRoot/deploy/testInstance/testBU/dataExtension/childBU_dataextension_test.dataExtension-meta.json:20]: this is real results im getting, so if they need removing, that should be managed in another PR.

File: test/resources/auth.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+    "success": {
+        "url": "https://mct0l7nxfq2r988t1kxfy8sc4xxx.auth.marketingcloudapis.com/v2/token",
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply test/resources/auth.json:3]: obfuscated yes, but can put a full dummy value if you prefer :)

File: lib/metadataTypes/Asset.js [hunk 5]
```diff
@@ -9,8 +9,8 @@ const Mustache = require('mustache');
 const cache = require('../util/cache');
 
 /**
- * @typedef {object.<string, any>} AssetItem
- * @typedef {object.<string, AssetItem>} AssetMap
+ * @typedef {Object<string, any>} AssetItem
+ * @typedef {Object<string, AssetItem>} AssetMap
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply lib/metadataTypes/Asset.js:13]: seems related to version, reverted version changes resolved this

File: lib/Retriever.js [hunk 6]
```diff
@@ -127,6 +127,7 @@ class Retriever {
                     }
                 }
             } catch (ex) {
+                console.log(ex);
```
[the remark this answers is not in the document]
Comment (DougMidgley) [reply lib/Retriever.js:130]: good catch :)

File: test/mockRoot/deploy/testInstance/testBU/dataExtension/childBU_dataextension_test.dataExtension-meta.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (DougMidgley) [reply test/mockRoot/deploy/testInstance/testBU/dataExtension/childBU_dataextension_test.dataExtension-meta.json:20]: double check as in deploy

**Pull request merged**